hera_qm/uvflag.py

```diff
@@ -630,20 +668,36 @@ def to_flag(self):
         elif self.mode == 'metric':
             self.flag_array = np.zeros_like(self.metric_array, dtype=np.bool)
             self.mode = 'flag'
+            self.weights_array = np.ones_like(self.metric_array, dtype=np.float)
         else:
             raise ValueError('Unknown UVFlag mode: ' + self.mode + '. Cannot convert to flag.')
         self.history += 'Converted to mode "flag" with ' + hera_qm_version_str
         self.clear_unused_attributes()
 
-    def to_metric(self):
-        '''Convert to metric mode. NOT SMART. Simply removes flag_array and initializes
-        metric_array with zeros.
+    def to_metric(self, convert_wgts=False):
+        '''Convert to metric mode. NOT SMART. Simply recasts flag_array as float
+        and uses this as the metric array.
+
+        Args:
+            convert_wgts : bool, if True convert self.weights_array to ones
+                unless a column or row is completely flagged, in which case
+                convert those pixels to zero.
         '''
         if self.mode == 'metric':
             return
         elif self.mode == 'flag':
-            self.metric_array = np.zeros_like(self.flag_array, dtype=np.float)
+            self.metric_array = self.flag_array.astype(np.float)
             self.mode = 'metric'
+            if convert_wgts:
+                self.weights_array = np.ones_like(self.weights_array)
+                if self.type == 'waterfall':
+                    for i, pol in enumerate(self.polarization_array):
+                        self.weights_array[:, :, i] *= ~qm_utils.and_rows_cols(self.flag_array[:, :, i])
+                elif self.type == 'baseline':
```

Comment:
Do we not care about self.type == 'antenna'?

Reply:
added